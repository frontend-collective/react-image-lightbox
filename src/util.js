/**
 * Placeholder for future translate functionality
 */
export function translate(str, replaceStrings = null) {
  if (!str) {
    return '';
  }

  let translated = str;
  if (replaceStrings) {
    Object.keys(replaceStrings).forEach(placeholder => {
      translated = translated.replace(placeholder, replaceStrings[placeholder]);
    });
  }

  return translated;
}

export function getWindowWidth() {
  return typeof global.window !== 'undefined' ? global.window.innerWidth : 0;
}

export function getWindowHeight() {
  return typeof global.window !== 'undefined' ? global.window.innerHeight : 0;
}

const isCrossOriginFrame = () => {
  try {
    return (
      global.window.location.hostname !== global.window.parent.location.hostname
    );
  } catch (e) {
    return true;
  }
};

// Get the highest window context that isn't cross-origin
// (When in an iframe)
export function getHighestSafeWindowContext(self = global.window.self) {
  // If we reached the top level, return self
  if (self === global.window.top) {
    return self;
  }

  // If parent is the same origin, we can move up one context
  // Reference: https://stackoverflow.com/a/21965342/1601953
  if (!isCrossOriginFrame()) {
    return getHighestSafeWindowContext(self.parent);
  }

  // If a different origin, we consider the current level
  // as the top reachable one
  return self;
}
