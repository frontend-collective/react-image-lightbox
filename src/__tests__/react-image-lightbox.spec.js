import { mount } from 'enzyme';
import React from 'react';
import Modal from 'react-modal';
import Lightbox from '../index';
import { translate, getHighestSafeWindowContext } from '../util';
import {
  MAX_ZOOM_LEVEL,
  MIN_ZOOM_LEVEL,
  ZOOM_BUTTON_INCREMENT_SIZE,
} from '../constant';

// Mock the loadStyles static function to avoid
// issues with a lack of styles._insertCss
Lightbox.loadStyles = jest.fn();

const commonProps = {
  mainSrc: '/fake/image/src.jpg',
  onCloseRequest: () => {},
};

const extendedCommonProps = {
  ...commonProps,
  prevSrc: '/fake/image/src1.jpg',
  nextSrc: '/fake/image/src2.jpg',
};

describe('Lightbox structure', () => {
  const wrapper = mount(<Lightbox {...commonProps} />);

  it('contains react-modal', () => {
    expect(wrapper.find(Modal).length).toEqual(1);
  });

  it('does not contain prev button when no prevSrc supplied', () => {
    expect(wrapper.find('.ril-prev-button').length).toEqual(0);
  });

  it('contains prev button when prevSrc supplied', () => {
    wrapper.setProps({ prevSrc: '/my/prev/src' });
    expect(wrapper.find('.ril-prev-button').length).toEqual(1);
  });

  it('does not contain next button when no nextSrc supplied', () => {
    expect(wrapper.find('.ril-next-button').length).toEqual(0);
  });

  it('contains next button when nextSrc supplied', () => {
    wrapper.setProps({ nextSrc: '/my/next/src' });
    expect(wrapper.find('.ril-next-button').length).toEqual(1);
  });

  it('contains zoom buttons when enableZoom is true (default)', () => {
    expect(wrapper.find('.ril-zoom-out').length).toEqual(1);
    expect(wrapper.find('.ril-zoom-in').length).toEqual(1);
  });

  it('does not contain zoom buttons when enableZoom is false', () => {
    wrapper.setProps({ enableZoom: false });
    expect(wrapper.find('.ril-zoom-out').length).toEqual(0);
    expect(wrapper.find('.ril-zoom-in').length).toEqual(0);
  });

  it('does not contain a caption bar when no caption is supplied', () => {
    expect(wrapper.find('.ril-caption').length).toEqual(0);
  });

  it('contains a caption bar when a caption is supplied', () => {
    wrapper.setProps({ caption: 'My Caption' });
    expect(wrapper.find('.ril-caption').length).toEqual(0);
  });

  it('contains custom toolbar buttons when supplied', () => {
    wrapper.setProps({
      toolbarButtons: [<button type="button" className="my-test-button" />],
    });
    expect(wrapper.find('.ril-toolbar__item .my-test-button').length).toEqual(
      1
    );
  });

  it('contains image title when supplied', () => {
    wrapper.setProps({
      imageTitle: <div className="my-image-title" />,
    });

    expect(
      wrapper.find(
        '.ril-toolbar-left .ril-toolbar__item__child .my-image-title'
      ).length
    ).toEqual(1);
  });
});

describe('Events', () => {
  const LOAD_FAILURE_SRC = 'LOAD_FAILURE_SRC';
  const LOAD_SUCCESS_SRC = 'LOAD_SUCCESS_SRC';
  let originalImageSrcProto;
  beforeAll(() => {
    originalImageSrcProto = Object.getOwnPropertyDescriptor(
      global.Image.prototype,
      'src'
    );

    Object.defineProperty(global.Image.prototype, 'src', {
      set(src) {
        if (src === LOAD_FAILURE_SRC) {
          setTimeout(() => this.onerror(new Error('mock error')));
        } else if (src === LOAD_SUCCESS_SRC) {
          setTimeout(this.onload);
        }
      },
    });
  });

  afterAll(() => {
    Object.defineProperty(global.Image.prototype, 'src', originalImageSrcProto);
  });

  const mockFns = {
    onAfterOpen: jest.fn(),
    onCloseRequest: jest.fn(),
    onMovePrevRequest: jest.fn(),
    onMoveNextRequest: jest.fn(),
    onImageLoad: jest.fn(),
    onImageLoadError: jest.fn(),
  };

  const wrapper = mount(
    <Lightbox {...extendedCommonProps} {...mockFns} animationDisabled />
  );

  // Spy zoomBtn focus
  const { zoomOutBtn, zoomInBtn } = wrapper.instance();
  jest.spyOn(zoomOutBtn.current, 'focus');
  jest.spyOn(zoomInBtn.current, 'focus');

  it('Calls onAfterOpen when mounted', async () => {
    // Rough way to wait for react-modal to call its onAfterOpen,
    // which is delayed by a requestAnimationFrame:
    // https://github.com/reactjs/react-modal/blob/fb6bab5e7/src/components/ModalPortal.js#L226-L230
    await new Promise(resolve => setTimeout(resolve));

    expect(mockFns.onAfterOpen).toHaveBeenCalledTimes(1);
    expect(mockFns.onAfterOpen).toHaveBeenCalledWith();
  });

  it('Calls onMovePrevRequest when left button clicked', () => {
    expect(mockFns.onMovePrevRequest).toHaveBeenCalledTimes(0);
    wrapper.find('.ril-prev-button').simulate('click');
    expect(mockFns.onMovePrevRequest).toHaveBeenCalledTimes(1);
    expect(mockFns.onMovePrevRequest).not.toHaveBeenCalledWith();
  });

  it('Calls onMoveNextRequest when right button clicked', () => {
    expect(mockFns.onMoveNextRequest).toHaveBeenCalledTimes(0);
    wrapper.find('.ril-next-button').simulate('click');
    expect(mockFns.onMoveNextRequest).toHaveBeenCalledTimes(1);
    expect(mockFns.onMoveNextRequest).not.toHaveBeenCalledWith();
  });

  it('Calls onCloseRequest when close button clicked', () => {
    expect(mockFns.onCloseRequest).toHaveBeenCalledTimes(0);
    wrapper.find('.ril-close').simulate('click');
    expect(mockFns.onCloseRequest).toHaveBeenCalledTimes(1);
    expect(mockFns.onCloseRequest).not.toHaveBeenCalledWith();
  });

  it('Calls onImageLoad when image loaded', done => {
    mockFns.onImageLoad.mockImplementationOnce((imageSrc, srcType, image) => {
      expect(imageSrc).toEqual(LOAD_SUCCESS_SRC);
      expect(srcType).toEqual('mainSrc');
      expect(image).toBeInstanceOf(global.Image);
      done();
    });

    expect(mockFns.onImageLoad).toHaveBeenCalledTimes(0);
    wrapper.setProps({ mainSrc: LOAD_SUCCESS_SRC });
  });

  it('Calls onImageLoadError when image loaded', done => {
    mockFns.onImageLoadError.mockImplementationOnce(
      (imageSrc, srcType, image) => {
        expect(imageSrc).toEqual(LOAD_FAILURE_SRC);
        expect(srcType).toEqual('mainSrc');
        expect(image).toBeInstanceOf(Error);
        done();
      }
    );

    expect(mockFns.onImageLoadError).toHaveBeenCalledTimes(0);
    wrapper.setProps({ mainSrc: LOAD_FAILURE_SRC });
  });

  it('Calls the the ZoomIn Focus when ZoomOut is disabled', () => {
    wrapper.setState({
      zoomLevel: MIN_ZOOM_LEVEL + ZOOM_BUTTON_INCREMENT_SIZE,
    });
    wrapper.instance().handleZoomOutButtonClick();
    expect(zoomInBtn.current.focus).toHaveBeenCalledTimes(1);
  });

  it('Calls the the ZoomOut Focus when ZoomIn is disabled', () => {
    wrapper.setState({
      zoomLevel: MAX_ZOOM_LEVEL - ZOOM_BUTTON_INCREMENT_SIZE,
    });
    wrapper.instance().handleZoomInButtonClick();
    expect(zoomOutBtn.current.focus).toHaveBeenCalledTimes(1);
  });
});

describe('Key bindings', () => {
  const mockCloseRequest = jest.fn();
  const mockMovePrevRequest = jest.fn();
  const mockMoveNextRequest = jest.fn();

  const wrapper = mount(
    <Lightbox
      {...commonProps}
      onCloseRequest={mockCloseRequest}
      onMovePrevRequest={mockMovePrevRequest}
      onMoveNextRequest={mockMoveNextRequest}
    />
  );

  const simulateKey = keyCode => {
    // Avoid interference by key throttling
    wrapper.instance().lastKeyDownTime = new Date('1970-01-01').getTime();

    wrapper.find('.ril-outer').simulate('keyDown', { keyCode });
  };

  it('Responds to close key binding', () => {
    expect(mockCloseRequest).toHaveBeenCalledTimes(0);
    // Simulate ESC key press
    simulateKey(27);
    expect(mockCloseRequest).toHaveBeenCalledTimes(1);
  });

  it('Doesn\'t respond to "move to next" key binding when no next image available', () => {
    // Simulate right arrow key press
    simulateKey(39);
    expect(mockMoveNextRequest).toHaveBeenCalledTimes(0);
  });

  it('Responds to "move to next" key binding when next image available', () => {
    wrapper.setProps({ nextSrc: '/my/next/src' });

    // Simulate right arrow key press
    simulateKey(39);
    expect(mockMoveNextRequest).toHaveBeenCalledTimes(1);
  });

  it('Doesn\'t respond to "move to prev" key binding when no prev image available', () => {
    // Simulate left arrow key press
    simulateKey(37);
    expect(mockMovePrevRequest).toHaveBeenCalledTimes(0);
  });

  it('Responds to "move to prev" key binding', () => {
    wrapper.setProps({ prevSrc: '/my/prev/src' });

    // Simulate left arrow key press
    simulateKey(37);
    expect(mockMovePrevRequest).toHaveBeenCalledTimes(1);
  });
});

describe('Snapshot Testing', () => {
  it('Lightbox renders properly"', () => {
    const wrapper = mount(
      <Lightbox
        {...commonProps}
        reactModalProps={{ appElement: global.document.createElement('div') }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Error Testing', () => {
  it('Should render the default error message', () => {
    const wrapper = mount(<Lightbox {...commonProps} />);
    wrapper.setState({
      loadErrorStatus: { mainSrc: true },
    });
    wrapper.update();
    expect(wrapper.find('div.ril__errorContainer')).toHaveText(
      'This image failed to load'
    );
  });
  it('Should render the specified error message', () => {
    const wrapper = mount(<Lightbox {...commonProps} />);
    const imageLoadErrorMessage = <p>Specified Error Message</p>;
    wrapper.setState({
      loadErrorStatus: { mainSrc: true },
    });
    wrapper.setProps({
      imageLoadErrorMessage,
    });
    wrapper.update();
    expect(wrapper.find('div.ril__errorContainer')).toContainReact(
      imageLoadErrorMessage
    );
  });
});

describe('Utils', () => {
  it('translate function return empty string if str param is not passed', () => {
    expect(translate()).toBe('');
  });
  it('getHighestSafeWindowContext function if parent is the same origin', () => {
    const self = {
      location: { href: 'http://test.test' },
      document: { referrer: 'http://test.test' },
    };
    expect(getHighestSafeWindowContext(self)).toBe(global.window.top);
  });
  it.skip('getHighestSafeWindowContext function if parent is a different origin', () => {
    const self = {
      location: { href: 'http://test1.test' },
      document: { referrer: 'http://test.test' },
    };
    expect(getHighestSafeWindowContext(self)).toBe(self);
  });
});
