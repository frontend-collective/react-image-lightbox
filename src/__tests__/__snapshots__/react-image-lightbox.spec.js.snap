// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Lightbox renders properly" 1`] = `
<ReactImageLightbox
  animationDisabled={false}
  animationDuration={300}
  animationOnKeyInput={false}
  clickOutsideToClose={true}
  closeLabel="Close lightbox"
  discourageDownloads={false}
  enableZoom={true}
  imageCaption={null}
  imageCrossOrigin={null}
  imageLoadErrorMessage="This image failed to load"
  imagePadding={10}
  imageTitle={null}
  keyRepeatKeyupBonus={40}
  keyRepeatLimit={180}
  mainSrc="/fake/image/src.jpg"
  mainSrcThumbnail={null}
  nextLabel="Next image"
  nextSrc={null}
  nextSrcThumbnail={null}
  onAfterOpen={[Function]}
  onCloseRequest={[Function]}
  onImageLoad={[Function]}
  onImageLoadError={[Function]}
  onMoveNextRequest={[Function]}
  onMovePrevRequest={[Function]}
  prevLabel="Previous image"
  prevSrc={null}
  prevSrcThumbnail={null}
  reactModalProps={
    Object {
      "appElement": <div
        aria-hidden="true"
      />,
    }
  }
  reactModalStyle={Object {}}
  toolbarButtons={null}
  wrapperClassName=""
  zoomInLabel="Zoom in"
  zoomOutLabel="Zoom out"
>
  <Modal
    appElement={
      <div
        aria-hidden="true"
      />
    }
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentElement={[Function]}
    contentLabel="Lightbox"
    isOpen={true}
    onAfterOpen={[Function]}
    onRequestClose={[Function]}
    overlayElement={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    preventScroll={false}
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "backgroundColor": "transparent",
          "border": "none",
          "borderRadius": 0,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "padding": 0,
          "right": 0,
          "top": 0,
        },
        "overlay": Object {
          "backgroundColor": "transparent",
          "zIndex": 1000,
        },
      }
    }
  >
    <Portal
      containerInfo={
        <div
          class="ReactModalPortal"
        >
          <div
            class="ReactModal__Overlay"
            style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: transparent; z-index: 1000;"
          >
            <div
              aria-label="Lightbox"
              aria-modal="true"
              class="ReactModal__Content"
              role="dialog"
              style="position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; background: rgb(255, 255, 255); overflow: hidden; border-radius: 0; outline: none; padding: 0px; background-color: transparent;"
              tabindex="-1"
            >
              <div
                class="ril-outer ril__outer ril__outerAnimating  "
                style="transition: opacity 300ms; animation-duration: 300ms; animation-direction: reverse;"
                tabindex="-1"
              >
                <div
                  class="ril-inner ril__inner"
                >
                  <div
                    class="ril-image-current ril__image ril-not-loaded"
                    style="transform: translate3d(0px,0px,0) scale3d(NaN,NaN,1);"
                  >
                    <div
                      class="ril__loadingContainer"
                    >
                      <div
                        class="ril-loading-circle ril__loadingCircle ril__loadingContainer__icon"
                      >
                        <div
                          class="ril-loading-circle-point ril__loadingCirclePoint"
                        />
                        <div
                          class="ril-loading-circle-point ril__loadingCirclePoint"
                        />
                        <div
                          class="ril-loading-circle-point ril__loadingCirclePoint"
                        />
                        <div
                          class="ril-loading-circle-point ril__loadingCirclePoint"
                        />
                        <div
                          class="ril-loading-circle-point ril__loadingCirclePoint"
                        />
                        <div
                          class="ril-loading-circle-point ril__loadingCirclePoint"
                        />
                        <div
                          class="ril-loading-circle-point ril__loadingCirclePoint"
                        />
                        <div
                          class="ril-loading-circle-point ril__loadingCirclePoint"
                        />
                        <div
                          class="ril-loading-circle-point ril__loadingCirclePoint"
                        />
                        <div
                          class="ril-loading-circle-point ril__loadingCirclePoint"
                        />
                        <div
                          class="ril-loading-circle-point ril__loadingCirclePoint"
                        />
                        <div
                          class="ril-loading-circle-point ril__loadingCirclePoint"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ril-toolbar ril__toolbar"
                >
                  <ul
                    class="ril-toolbar-left ril__toolbarSide ril__toolbarLeftSide"
                  >
                    <li
                      class="ril-toolbar__item ril__toolbarItem"
                    >
                      <span
                        class="ril-toolbar__item__child ril__toolbarItemChild"
                      />
                    </li>
                  </ul>
                  <ul
                    class="ril-toolbar-right ril__toolbarSide ril__toolbarRightSide"
                  >
                    <li
                      class="ril-toolbar__item ril__toolbarItem"
                    >
                      <button
                        aria-label="Zoom in"
                        class="ril-zoom-in ril__toolbarItemChild ril__builtinButton ril__zoomInButton"
                        title="Zoom in"
                        type="button"
                      />
                    </li>
                    <li
                      class="ril-toolbar__item ril__toolbarItem"
                    >
                      <button
                        aria-label="Zoom out"
                        class="ril-zoom-out ril__toolbarItemChild ril__builtinButton ril__zoomOutButton ril__builtinButtonDisabled"
                        disabled=""
                        title="Zoom out"
                        type="button"
                      />
                    </li>
                    <li
                      class="ril-toolbar__item ril__toolbarItem"
                    >
                      <button
                        aria-label="Close lightbox"
                        class="ril-close ril-toolbar__item__child ril__toolbarItemChild ril__builtinButton ril__closeButton"
                        title="Close lightbox"
                        type="button"
                      />
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      }
    >
      <ModalPortal
        appElement={
          <div
            aria-hidden="true"
          />
        }
        ariaHideApp={true}
        bodyOpenClassName="ReactModal__Body--open"
        closeTimeoutMS={0}
        contentElement={[Function]}
        contentLabel="Lightbox"
        defaultStyles={
          Object {
            "content": Object {
              "WebkitOverflowScrolling": "touch",
              "background": "#fff",
              "border": "1px solid #ccc",
              "borderRadius": "4px",
              "bottom": "40px",
              "left": "40px",
              "outline": "none",
              "overflow": "auto",
              "padding": "20px",
              "position": "absolute",
              "right": "40px",
              "top": "40px",
            },
            "overlay": Object {
              "backgroundColor": "rgba(255, 255, 255, 0.75)",
              "bottom": 0,
              "left": 0,
              "position": "fixed",
              "right": 0,
              "top": 0,
            },
          }
        }
        isOpen={true}
        onAfterOpen={[Function]}
        onRequestClose={[Function]}
        overlayElement={[Function]}
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        preventScroll={false}
        role="dialog"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
        style={
          Object {
            "content": Object {
              "backgroundColor": "transparent",
              "border": "none",
              "borderRadius": 0,
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "padding": 0,
              "right": 0,
              "top": 0,
            },
            "overlay": Object {
              "backgroundColor": "transparent",
              "zIndex": 1000,
            },
          }
        }
      >
        <div
          className="ReactModal__Overlay"
          onClick={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "fixed",
              "right": 0,
              "top": 0,
              "zIndex": 1000,
            }
          }
        >
          <div
            aria-label="Lightbox"
            aria-modal={true}
            className="ReactModal__Content"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            role="dialog"
            style={
              Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "backgroundColor": "transparent",
                "border": "none",
                "borderRadius": 0,
                "bottom": 0,
                "left": 0,
                "outline": "none",
                "overflow": "hidden",
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            tabIndex="-1"
          >
            <div
              className="ril-outer ril__outer ril__outerAnimating  "
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseMove={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              onWheel={[Function]}
              style={
                Object {
                  "animationDirection": "reverse",
                  "animationDuration": "300ms",
                  "transition": "opacity 300ms",
                }
              }
              tabIndex="-1"
            >
              <div
                className="ril-inner ril__inner"
                onClick={[Function]}
              >
                <div
                  className="ril-image-current ril__image ril-not-loaded"
                  key="/fake/image/src.jpgi0"
                  style={
                    Object {
                      "transform": "translate3d(0px,0px,0) scale3d(NaN,NaN,1)",
                    }
                  }
                >
                  <div
                    className="ril__loadingContainer"
                  >
                    <div
                      className="ril-loading-circle ril__loadingCircle ril__loadingContainer__icon"
                    >
                      <div
                        className="ril-loading-circle-point ril__loadingCirclePoint"
                        key="0"
                      />
                      <div
                        className="ril-loading-circle-point ril__loadingCirclePoint"
                        key="1"
                      />
                      <div
                        className="ril-loading-circle-point ril__loadingCirclePoint"
                        key="2"
                      />
                      <div
                        className="ril-loading-circle-point ril__loadingCirclePoint"
                        key="3"
                      />
                      <div
                        className="ril-loading-circle-point ril__loadingCirclePoint"
                        key="4"
                      />
                      <div
                        className="ril-loading-circle-point ril__loadingCirclePoint"
                        key="5"
                      />
                      <div
                        className="ril-loading-circle-point ril__loadingCirclePoint"
                        key="6"
                      />
                      <div
                        className="ril-loading-circle-point ril__loadingCirclePoint"
                        key="7"
                      />
                      <div
                        className="ril-loading-circle-point ril__loadingCirclePoint"
                        key="8"
                      />
                      <div
                        className="ril-loading-circle-point ril__loadingCirclePoint"
                        key="9"
                      />
                      <div
                        className="ril-loading-circle-point ril__loadingCirclePoint"
                        key="10"
                      />
                      <div
                        className="ril-loading-circle-point ril__loadingCirclePoint"
                        key="11"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="ril-toolbar ril__toolbar"
              >
                <ul
                  className="ril-toolbar-left ril__toolbarSide ril__toolbarLeftSide"
                >
                  <li
                    className="ril-toolbar__item ril__toolbarItem"
                  >
                    <span
                      className="ril-toolbar__item__child ril__toolbarItemChild"
                    />
                  </li>
                </ul>
                <ul
                  className="ril-toolbar-right ril__toolbarSide ril__toolbarRightSide"
                >
                  <li
                    className="ril-toolbar__item ril__toolbarItem"
                  >
                    <button
                      aria-label="Zoom in"
                      className="ril-zoom-in ril__toolbarItemChild ril__builtinButton ril__zoomInButton"
                      disabled={false}
                      key="zoom-in"
                      onClick={[Function]}
                      title="Zoom in"
                      type="button"
                    />
                  </li>
                  <li
                    className="ril-toolbar__item ril__toolbarItem"
                  >
                    <button
                      aria-label="Zoom out"
                      className="ril-zoom-out ril__toolbarItemChild ril__builtinButton ril__zoomOutButton ril__builtinButtonDisabled"
                      disabled={true}
                      key="zoom-out"
                      title="Zoom out"
                      type="button"
                    />
                  </li>
                  <li
                    className="ril-toolbar__item ril__toolbarItem"
                  >
                    <button
                      aria-label="Close lightbox"
                      className="ril-close ril-toolbar__item__child ril__toolbarItemChild ril__builtinButton ril__closeButton"
                      key="close"
                      onClick={[Function]}
                      title="Close lightbox"
                      type="button"
                    />
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </ModalPortal>
    </Portal>
  </Modal>
</ReactImageLightbox>
`;
